libbeat/outputs/shipper/shipper.go

```diff
@@ -239,3 +215,49 @@ func convertMapStr(m mapstr.M) (*structpb.Value, error) {
 
 	return structpb.NewStructValue(s), nil
 }
+
+func toShipperEvent(e publisher.Event) (*sc.Event, error) {
```

Comment:
Not for this review, but it would be interesting to benchmark the conversions to the shipper data types to see what the overhead is. This will be new overhead added to every event sent from every beat.